### PR TITLE
docs: aws marketplace

### PR DESCRIPTION
**Pull request by chri2547**
**What this PR does / why we need it**: These docs support the AWS Marketplace rollout.

**Which issue(s) this PR fixes**:

Fixes grafana/grafana-enterprise#2476
Fixes grafana/grafana-enterprise-partnerships-team#38

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/enterprise/license/_index.md [hunk 1]
```diff
@@ -7,9 +7,10 @@ weight = 10
 
 # Grafana Enterprise license
 
-To run Grafana Enterprise, you need a valid license. [Contact a Grafana Labs representative](https://grafana.com/contact?about=grafana-enterprise) to obtain the license. For information on how to activate your license, refer to [Activate an Enterprise license]({{< relref "./activate-license.md" >}}).
+When you become a Grafana Enterprise customer, you gain access to Grafans's premimum observability features, including enterprise data source plugins, reporting, and fine-grained access control. In order to use these [enhanced features of Grafana Enterprise]({{< relref "../_index.md" >}}), you must purchase and activate a Grafana Enterprise license.
```
Comment (mjseaman) [inline docs/sources/enterprise/license/_index.md:10]:
```suggestion
When you become a Grafana Enterprise customer, you gain access to Grafana's premium observability features, including enterprise data source plugins, reporting, and fine-grained access control. In order to use these [enhanced features of Grafana Enterprise]({{< relref "../_index.md" >}}), you must purchase and activate a Grafana Enterprise license.
```

File: docs/sources/enterprise/license/activate-aws-marketplace-license/about-ge-license-through-aws.md [hunk 2]
```diff
@@ -8,19 +8,19 @@ weight = 100
...
 # About Grafana Enterprise licenses from AWS Marketplace
 
-AWS Marketplace is a convenient place to buy and manage a Grafana Enterprise license, especially if you are already an AWS customer.
+AWS Marketplace is a convenient place for AWS customers to buy and manage a license for Grafana Enterprise versions 8.3.0 and later.
 
 You can deploy Grafana Enterprise in the following ways:
 
-- You can deploy Grafana Enterprise in AWS ECS or AWS EKS, or,
-- You can deploy Grafana Enterprise on a Grafana Enterprise instance outside AWS.
+- In an instance of AWS ECS or AWS EKS.
```
Comment (guicaulada) [inline docs/sources/enterprise/license/activate-aws-marketplace-license/about-ge-license-through-aws.md:15]: I believe it might be better to change this to:
```suggestion
- Using AWS services like ECS, EKS or EC2.
```
Since any AWS sevice can be used to deploy Grafana using this license.

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md [hunk 3]
```diff
@@ -10,70 +10,95 @@ weight = 250
...
-```
+### Before you begin
 
-> For more information about how to update your ECS service with an environment variable, refer to [Updating a service using the new console](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/update-service-console-v2.html).
+- Ensure that you have a MySQL database available. For information about creating a MySQL database, refer to [Creating an Amazon RDS DB instance](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_CreateDBInstance.html).
+- Review the information required to connect to the RDS DB instance. For more information, refer to [Connecting to an Amazon RDS DB instance](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_CommonTasks.Connect.html).
+
+To configure Grafana for high availability:
+
+1. In Grafana Enterprise, open the grafana.ini file and update the [database] parameters.
```
Comment (guicaulada) [inline docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md:43]: On ECS these parameters can only be updated using the environment variables, they won't be able to update the configuration file inside the container, the link to the configuration was just for reference on the possible values for the parameters of the database section.

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md [hunk 4]
```diff
@@ -10,70 +10,95 @@ weight = 250
...
-GF_DATABASE_TYPE=mysql
-GF_DATABASE_HOST=localhost:3306
-GF_DATABASE_NAME=grafana
-GF_DATABASE_USER=[mysql database username]
-GF_DATABASE_PASSWORD=[mysql database password]
-```
+### Before you begin
 
-> For more information about how to update your ECS service with an environment variable, refer to [Updating a service using the new console](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/update-service-console-v2.html).
+- Ensure that you have a MySQL database available. For information about creating a MySQL database, refer to [Creating an Amazon RDS DB instance](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_CreateDBInstance.html).
```
Comment (guicaulada) [inline docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md:38]: The database doesn't necesseraly need to be MySQL, it can be any of the supported types by Grafana, at this moment MySQL or Postgres.

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md [hunk 5]
```diff
@@ -10,70 +10,95 @@ weight = 250
...
+   For more information about configuring the database, refer to [Configuration]({{< relref "../../../administration/configuration.md#database" >}}).
+
+1. Create a revision of the task definition for the ECS Task that runs Grafana Enterprise.
+
+   For more information about creating a task, refer to [Updating a task definition using the classic console](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/update-task-definition.html).
+
+1. Within the new revision, edit the Grafana Enterprise container for this task, and add the following environment variable to the container:
+
+   ```
+   GF_DATABASE_TYPE=mysql
```
Comment (guicaulada) [inline docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md:54]: Maybe we should use `[database type]`

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md [hunk 6]
```diff
@@ -10,70 +10,95 @@ weight = 250
...
+   GF_DATABASE_HOST=localhost:3306
```
Comment (guicaulada) [inline docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md:55]: Maybe we should use `[database address and port]`

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md [hunk 7]
```diff
@@ -10,70 +10,95 @@ weight = 250
...
+   GF_DATABASE_NAME=grafana
```
Comment (guicaulada) [inline docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md:56]: MAybe we hsould use `[database name]`

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md [hunk 8]
```diff
@@ -10,70 +10,95 @@ weight = 250
...
+   GF_DATABASE_USER=[mysql database username]
```
Comment (guicaulada) [inline docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md:57]: This could be changed to `[database username]` since the database can be mysql or postgres.

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md [hunk 9]
```diff
@@ -10,70 +10,95 @@ weight = 250
...
+   GF_DATABASE_PASSWORD=[mysql database password]
```
Comment (guicaulada) [inline docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md:58]: This could be changed to `[database password]` since the database can be mysql or postgres.

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-eks.md [hunk 10]
```diff
@@ -19,77 +19,89 @@ To activate your license, complete the following tasks:
...
-type = mysql
-host = localhost:3306
-name = grafana
-user = [mysql database username]
-password = [mysql database password]
-```
+### Before you begin
 
-**Option 2:** Use `kubectl edit deployment my-release` to edit the pod environment variables. Under `env` add the following variables:
+- Ensure that you have a MySQL database available. For information about creating a MySQL database, refer to [Creating an Amazon RDS DB instance](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_CreateDBInstance.html).
```
Comment (guicaulada) [inline docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-eks.md:40]: Same as for ECS, the database can by MySQL or Postgres.

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-eks.md [hunk 11]
```diff
@@ -19,77 +19,89 @@ To activate your license, complete the following tasks:
...
-  value: [mysql database username]
-- name: GF_DATABASE_PASSWORD
-  value: [mysql database password]
-```
+To configure Grafana for high availability, perform one of the following steps:
+
+- Use `kubectl edit configmap grafana` to edit `grafana.ini` add the following section to the configuration:
+
+  ```
+  [database]
```
Comment (guicaulada) [inline docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-eks.md:48]: Same as for ECS maybe the values in this section could be changed to placeholder values.

On EKS it's possible to update the configuration file since it's a ConfigMap and not inside the container.

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-eks.md [hunk 12]
```diff
@@ -19,77 +19,89 @@ To activate your license, complete the following tasks:
...
+  type = mysql
+  host = localhost:3306
+  name = grafana
+  user = [mysql database username]
+  password = [mysql database password]
+  ```
+
+- Use `kubectl edit deployment my-release` to edit the pod `env` variables and add the following database variables:
```
Comment (guicaulada) [inline docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-eks.md:56]: Same as above, maybe the values for `mysql`, `localhost:3306` and `grafana` could be placeholders.

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md (hunk 4 above)
Comment (chri2547) [reply docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md:38]: Took care of this one. Added a link to our list of supported dbs and removed references to MySQL.

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md (hunk 3 above)
Comment (chri2547) [reply docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md:43]: Got it, made that change.

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md (hunk 5 above)
Comment (chri2547) [reply docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md:54]: Done, please check.

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md (hunk 6 above)
Comment (chri2547) [reply docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md:55]: Done, please check.

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md (hunk 7 above)
Comment (chri2547) [reply docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md:56]: Done.

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md (hunk 8 above)
Comment (chri2547) [reply docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md:57]: done.

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md (hunk 9 above)
Comment (chri2547) [reply docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-ecs.md:58]: done.

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-eks.md (hunk 10 above)
Comment (chri2547) [reply docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-eks.md:40]: Done.

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-eks.md (hunk 11 above)
Comment (chri2547) [reply docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-eks.md:48]: made those changes

File: docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-eks.md (hunk 12 above)
Comment (chri2547) [reply docs/sources/enterprise/license/activate-aws-marketplace-license/activate-license-on-eks.md:56]: Done.

File: docs/sources/panels/transformations/_index.md [hunk 13]
```diff
@@ -29,7 +29,7 @@ Before you can configure and apply transformations:
 - [Add field from calculation]({{< relref "./types-options.md#add-field-from-calculation" >}})
 - [Concatenate fields]({{< relref "./types-options.md#concatenate-fields" >}})
 - [Config from query results]({{< relref "./config-from-query.md" >}})
-- [Convert field type]({{< relref "./convert-field-type.md" >}})
+- [Convert field type]({{< relref "./types-options.md#convert-field-type" >}})
```
Comment (guicaulada) [inline docs/sources/panels/transformations/_index.md:32]: This change is unrelated? I think it might be just a small fix? Just confirming it's something that should be here and not an unwanted change.

Otherwise, LGTM!

**Review by guicaulada — APPROVED** [review]

**Comment by mjseaman** [comment]
Just added a small number of copy edits, looking good!

Notes:
- The titles of each topic are quite long - they're hard to read in the docs sidebar. Should we care about this @chri2547 ?
- IMO we should put the content on the About page into the index page, and delete the About page. It seems redundant to have both but I defer to @chri2547.
- @guicaulada is it necessary to configure Grafana for High Availability on EKS, or is that an optional step? If optional, I vote that we remove it or at least note that it is optional. For ECS at least, I don't remember having to configure a database or HA...

**Comment by guicaulada** [comment]
It is necessary, otherwise Grafana will use sqlite and the token will be lost every time the container or pod is restarted on EKS or ECS, since the token is stored on the database.

File: docs/sources/panels/transformations/_index.md (hunk 13 above)
Comment (chri2547) [reply docs/sources/panels/transformations/_index.md:32]: Ah yes, small fix that I ran into as I worked through the docs.

**Comment by chri2547** [comment]
> Just added a small number of copy edits, looking good!
> 
> Notes:
> 
> * The titles of each topic are quite long - they're hard to read in the docs sidebar. Should we care about this @chri2547 ?
> * IMO we should put the content on the About page into the index page, and delete the About page. It seems redundant to have both but I defer to @chri2547.
> * @guicaulada is it necessary to configure Grafana for High Availability on EKS, or is that an optional step? If optional, I vote that we remove it or at least note that it is optional. For ECS at least, I don't remember having to configure a database or HA...

1. Title length - I agree that they are long. Optimal length should be 7-12 words. Longer (but not too long) is better because a) it's descriptive enough for users to know what content they'll see when they click the link, and b) longer titles help orient users if they Google search and land on that page, c) topic titles are weighted more heavily than body content in search results. The downside is the left nav, which we should live with for now, though I'm open to suggestions. 

2. Removing About (and moving content to the index page) - also a function of our IA where we are required to have an index page for each topic area. And we (tech docs) still need to figure out how to handle this interstitial page. It's a bit awkward now and a known issue. The problem with removing the About content and placing it on the index page is a) this creates an inconsistency with our newly-defined standard of labeling conceptual material with the word 'About' and b) users can technically bypass the index page content by clicking the down chevron in the TOC. Let's leave for now.

**Review by mjseaman — APPROVED** [review]
LGTM!

**Comment by guicaulada** [comment]
I will be backporting manually.
